modules/core/src/directx.cpp

```diff
@@ -678,43 +759,65 @@ Context& initializeContextFromDirect3DDevice9(IDirect3DDevice9* pDirect3DDevice9
+#endif //HAVE_CUDA
 
 static void __OpenCLinitializeD3D11()
 {
-    using namespace cv::ocl;
-    static cl_platform_id initializedPlatform = NULL;
-    cl_platform_id platform = (cl_platform_id)Platform::getDefault().ptr();
-    if (initializedPlatform != platform)
-    {
-        clCreateFromD3D11Texture2DKHR = (clCreateFromD3D11Texture2DKHR_fn)
-                clGetExtensionFunctionAddressForPlatform(platform, "clCreateFromD3D11Texture2DKHR");
-        clEnqueueAcquireD3D11ObjectsKHR = (clEnqueueAcquireD3D11ObjectsKHR_fn)
-                clGetExtensionFunctionAddressForPlatform(platform, "clEnqueueAcquireD3D11ObjectsKHR");
-        clEnqueueReleaseD3D11ObjectsKHR = (clEnqueueReleaseD3D11ObjectsKHR_fn)
-                clGetExtensionFunctionAddressForPlatform(platform, "clEnqueueReleaseD3D11ObjectsKHR");
-        initializedPlatform = platform;
-    }
-    if (!clCreateFromD3D11Texture2DKHR || !clEnqueueAcquireD3D11ObjectsKHR || !clEnqueueReleaseD3D11ObjectsKHR)
-    {
-        CV_Error(cv::Error::OpenCLInitError, "OpenCL: Can't find functions for D3D11");
-    }
+	using namespace cv::ocl;
+	static cl_platform_id initializedPlatform = NULL;
+	cl_platform_id platform = (cl_platform_id)Platform::getDefault().ptr();
+#ifdef HAVE_CUDA
+	if (initializedPlatform != platform)
+	{
+		clCreateFromD3D11Texture2DNV = (clCreateFromD3D11Texture2DNV_fn)
+			clGetExtensionFunctionAddressForPlatform(platform, "clCreateFromD3D11Texture2DNV");
+		clEnqueueAcquireD3D11ObjectsNV = (clEnqueueAcquireD3D11ObjectsNV_fn)
+			clGetExtensionFunctionAddressForPlatform(platform, "clEnqueueAcquireD3D11ObjectsNV");
+		clEnqueueReleaseD3D11ObjectsNV = (clEnqueueReleaseD3D11ObjectsNV_fn)
+			clGetExtensionFunctionAddressForPlatform(platform, "clEnqueueReleaseD3D11ObjectsNV");
+		initializedPlatform = platform;
+	}
+	if (!clCreateFromD3D11Texture2DNV || !clEnqueueAcquireD3D11ObjectsNV || !clEnqueueReleaseD3D11ObjectsNV)
+	{
+		CV_Error(cv::Error::OpenCLInitError, "OpenCL: Can't find functions for D3D11");
+	}
+#else 
+	if (initializedPlatform != platform)
+	{
+		clCreateFromD3D11Texture2DKHR = (clCreateFromD3D11Texture2DKHR_fn)
+			clGetExtensionFunctionAddressForPlatform(platform, "clCreateFromD3D11Texture2DKHR");
+		clEnqueueAcquireD3D11ObjectsKHR = (clEnqueueAcquireD3D11ObjectsKHR_fn)
+			clGetExtensionFunctionAddressForPlatform(platform, "clEnqueueAcquireD3D11ObjectsKHR");
+		clEnqueueReleaseD3D11ObjectsKHR = (clEnqueueReleaseD3D11ObjectsKHR_fn)
+			clGetExtensionFunctionAddressForPlatform(platform, "clEnqueueReleaseD3D11ObjectsKHR");
+		initializedPlatform = platform;
+	}
+	if (!clCreateFromD3D11Texture2DKHR || !clEnqueueAcquireD3D11ObjectsKHR || !clEnqueueReleaseD3D11ObjectsKHR)
+	{
+		CV_Error(cv::Error::OpenCLInitError, "OpenCL: Can't find functions for D3D11");
+	}
+#endif// HAVE_CUDA
 }
-#endif // defined(HAVE_DIRECTX) && defined(HAVE_OPENCL)
+#endif// defined(HAVE_DIRECTX) &&  defined(HAVE_OPENCL)
 
 } // namespace directx
 
-
 namespace ocl {
```

Comment:
Please avoid unrelated changes.

Reply:
Just delete blank lines.

Reply:
No need to do that.

Blank lines are useful for git merges to avoid unnecessary conflicts.

We don't use line fillers, like `////////////////////////////////////////` due different number of used symbols.

Reply:
Okay